traversal/patch/patch_serial.go

```diff
@@ -0,0 +1,73 @@
+package patch
+
+import (
+	"io"
+	"strings"
+
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+
+	"github.com/ipld/go-ipld-prime/codec/json"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/schema"
+	schemadmt "github.com/ipld/go-ipld-prime/schema/dmt"
+	schemadsl "github.com/ipld/go-ipld-prime/schema/dsl"
+)
+
+var ts = func() schema.TypeSystem {
+	sch, err := schemadsl.Parse("", strings.NewReader(
+		// This could be more accurately modelled as an inline union,
+		// but that seems like work, given how high the overlap is.
+		`
+		type Operation struct {
+			op String
```

Comment:
String representation enum would be nice here, also an opportunity to consider whether all those operations really are that useful; I'm instinctively dubious about basically everything except `add` and `remove`, the rest feel like complexity for the sake of theoretical efficiency. Maybe there's a case but they really turn the whole logic set (and therefore test suite for this) into a complex tangle.

Reply:
Yeah, this might get schema'd harder; I whipped this off the current state of this quite quickly.  I should consider putting [bindnode.ProduceGoTypes](https://pkg.go.dev/github.com/ipld/go-ipld-prime/node/bindnode#ProduceGoTypes) to work.

The sheer number of operations: I admit I blinked at the JSON Patch spec at first too, but think they're all justified.  'replace' is actually needed, since 'add' is supposed to error rather than upsert if the target path exists.  'move' and 'copy' are a little more arguable, but can't really be replaced if the situation is that you aren't already holding the document contents.  (Consider using 'move' to shift content in a HAMT from under one map key to another.  You wouldn't want to download the document just to give it a different key.)

I think we may end with slightly more operations, eventually.  E.g. I think I'd like an 'add2' that will create parent paths without a fuss, and upsert by default (perhaps an optional bool field to disable upsert).  We may also need to add some features to describe the distinction between "update this document that is the root of a block that is linked to" vs "replace this link node with something else entirely".  But sticking with the JSON Patch RFC for a baseline still seems like an easy choice.

Reply:
Added the enum for the Op.

Should I be copying the schema into `ipld/ipld` somewhere as it gets updated?

Reply:
The `_ "embed"` trick is great for this!